### PR TITLE
Add Custom Quick Actions to the dashboard

### PR DESCRIPTION
**Pull request by CaveMobster**
### Description of the Change

This adds Custom Quick Actions to the dashboard. It includes settings to show and hide Quick Actions (including the default ones) and to change the order in the bar. Custom Quick Actions have a context menu to edit or delete them, and the edit modal has an icon picker to customize the look of the Quick Action. 

### Applicable Issues

https://github.com/crowbartools/Firebot/issues/1210

### Testing

I tested adding, deleting, editing, hiding, showing, changing order, and if all settings save correctly. I also tested if quick actions work as expected, and I tested the UI.

### Screenshots

image
image
image

### CODE REVIEW

**Review by ebiggz — COMMENTED** [review]
Great work on this! Absolutely love it.

Any reason why we don't allow the creation of effects directly inside the Quick Action? As it is designed currently, it will be the only place we force the use of Preset Lists. Some folks might want to use preset lists for sure but for others it would only serve as an extra hoop they gotta jump through and otherwise splits the configuration of the Quick Action into two different places. Preferably, we would have it like the "Run Effects" effect where you can choose to create a list or select a preset.

**Comment by CaveMobster** [comment]
@ebiggz You know what, I actually don't remember why I chose to use Preset Effect Lists. 😅 I think I had issues with only sending definitions to the frontend instead of both the definition and the event, but I'm honestly not sure if that was actually the case. I'll look into it!

**Comment by CaveMobster** [comment]
@ebiggz That was easier than I thought haha. I updated the PR description with a new screenshot.

**Review by ebiggz — APPROVED** [review]

File: gui/app/directives/modals/chat/add-or-edit-custom-quick-action-modal.js [hunk 1]
```diff
@@ -24,15 +24,26 @@
...
+                            <ui-select ng-model="$ctrl.quickAction.presetListId" theme="bootstrap" on-select="presetListSelected($item)">
+                                <ui-select-match placeholder="Select or search for a preset effect list... ">{{$select.selected.name}}</ui-select-match>
+                                <ui-select-choices repeat="presetList.id as presetList in $ctrl.presetEffectLists | filter: { name: $select.search }" style="position:relative;">
+                                    <div ng-bind-html="presetList.name | highlight: $select.search"></div>
+                                </ui-select-choices>
+                            </ui-select>
+                        </div>
+                        <div ng-if="$ctrl.listType === 'custom'" class="mt-8">
+                            <effect-list effects="$ctrl.quickAction.effectList"
+                                trigger="{{'quick_action'}}"
```
[the remark this answers is not in the document]
Comment (CaveMobster) [reply gui/app/directives/modals/chat/add-or-edit-custom-quick-action-modal.js]: Omg I always get confused about those things haha, thanks! I updated it.

**Review by ebiggz — APPROVED** [review]

**Pull request merged**